src/hooks/useEditorView.ts

```diff
@@ -117,12 +109,8 @@ export function useEditorView<T extends HTMLElement = HTMLElement>(
   }, [editorProps, mount, state, view]);
 
   useLayoutEffect(() => {
-    view?.setProps(nonStateProps);
-  }, [view, nonStateProps]);
-
-  useLayoutEffect(() => {
-    if (stateProp) view?.setProps({ state: stateProp });
-  }, [view, stateProp]);
+    view?.setProps(editorProps);
+  }, [view, editorProps]);
```

Comment:
I think this might cause subtle issues. If I remember correctly, ProseMirror will execute side effects when various props are set (e.g. destroying and recreating all node views when the nodeViews prop is set), even if the reference is the same. We separated out the state update because the state prop updates very often as a user edits the document, and we don't want to trigger these side effects every time the state is updated.

Reply:
Actually, I think that the failing test might even be an example of what I'm talking about here (it might be something else, though)

Reply:
I thought we worked with Marijn to fix some of those issues upstream, so that even if we set the props things should be efficient as long as the value of the individual props didn't change.

Reply:
If they remain separated, is the initial approach of setting the state props before the nonState props in the layout effect alright in this case?

Reply:
I think doing it all at once is the safest, so if we do need to be efficient we can do it like we have in production code before the upstreamed changes to prosemirror-view that fixed inefficiencies:

```
    const { state: oldState, ...oldProps } = view.props;
    const { state: newState, ...newProps } = props;
    if (!isShallowEqual(oldProps, newProps)) {
      view.update(props);
    } else {
      view.updateState(props.state);
    }
```

Reply:
Isn't the layout effect setup now essentially just doing that shallow equal comparison for us?

Reply:
... ha. I just realized that we're actually calling view.setProps on literally every render, right now, because nonStateProps isn't memoized. So actually I'm not sure @ilyaGurevich's initial change would have caused issues.

Also, I'm not sure why we use `setProps` here rather than `update`/`updateState`? @tilgovi do you remember?

Reply:
`update` and `updateState` allow partial updates, I think, but we have the whole props.

Reply:
Hmm ok, what's the preferred solution here? It looks like we have multiple approaches all relatively doing similar things:
1. rearrange order (with memoized nonStateProps)
```
  useLayoutEffect(() => {
    if (stateProp) view?.setProps({ state: stateProp });
  }, [view, stateProp]);
  
  useLayoutEffect(() => {
    view?.setProps(nonStateProps);
  }, [view, nonStateProps]);
  ```
  
2. Unify props setting
```
  useLayoutEffect(() => {
    view?.setProps(editorProps);
  }, [view, editorProps]);
```

3. Use the older version in production
```
    const { state: oldState, ...oldProps } = view.props;
    const { state: newState, ...newProps } = props;
    if (!isShallowEqual(oldProps, newProps)) {
      view.update(props);
    } else {
      view.updateState(props.state);
    }
```

Reply:
I think unified props setting is the most sensible, and I'd be curious about the test breakage.